backend/src/services/auto_trader.py

```
@@ -97,174 +89,195 @@ def stop_trading(self):
-        logger.info("AutoTrader: Monitoring owned tokens for sell opportunities...")
         tokens_to_remove = []
-        for token_address, details in self.owned_tokens.items():
+        for token_address, details in list(self.owned_tokens.items()):
             try:
-                current_token_data = await self.data_fetcher_service.get_token_by_address(token_address)
-                if not current_token_data:
-                    logger.warning(f"AutoTrader: Owned token {details.get('symbol', token_address)} not found in data fetcher. Skipping sell monitoring.")
+                # Real balance check
+                try:
+                    current_balance = await self.wallet_service.get_token_balance(token_address)
+                except Exception as balance_e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)